samples/sdl_joystick_gamepad/main.cpp

```diff
@@ -0,0 +1,50 @@
+#include <xboxrt/debug.h>
+#include <pbkit/pbkit.h>
+#include <hal/xbox.h>
+#include <assert.h>
+#include <SDL.h>
+
+#define ARRAY_SIZE(x) (sizeof(x)/sizeof(x[0]))
+
+int main() {
+  int ret = pb_init();
+  if (ret != 0) {
+    XSleep(2000);
+    return -1;
+  }
+
+  pb_show_debug_screen();
+
+  SDL_GameController *joy;
+  SDL_InitSubSystem(SDL_INIT_GAMECONTROLLER);
+
+  joy = SDL_GameControllerOpen(0);
+
+  SDL_Event event;
+
+  debugPrint("Num joysticks      : %02i\n", SDL_NumJoysticks());
+  debugPrint("GameController name: %s\n", SDL_GameControllerName(joy));
+  debugPrint("Is GameController  : %02i\n", SDL_IsGameController(0));
+  debugPrint("GameController Map : %s\n", SDL_GameControllerMapping(joy));
+  //debugPrint("JoystickEvent      : %02i\n", SDL_JoystickEventState(SDL_QUERY));
+  //debugPrint("GameControllerEvent: %02i\n", SDL_GameControllerEventState(SDL_DISABLE));
+  //debugPrint("JoystickEvent      : %02i\n", SDL_JoystickEventState(SDL_DISABLE));
+  debugPrint("GameControllerEvent: %02i\n", SDL_GameControllerEventState(SDL_ENABLE));
```

Comment:
This seems to be a setter, that's being used like a getter.

`debugPrint` shouldn't be called with sideeffects, because it might be removed in real applications (so it shouldn't have sideeffects).